### PR TITLE
GitHub Integration

### PR DESCRIPTION
**Pull request by gokmen**
## Enable Github Integration for Team (https://github.com/koding/koding/issues/9778)

full screencast: http://recordit.co/71ZcWDEYPy

## Enable Github/GitLab integration for user (https://github.com/koding/koding/issues/9780)

full screencast: http://recordit.co/voUqQ7ocdu

## Login with Github (https://github.com/koding/koding/issues/9779)

full screencast: http://recordit.co/w27g6AIdvq

## Both integrations enabled views

image

## Scope support for both team and user integration

full screencast: http://recordit.co/fF3UVUEtjV

### CODE REVIEW

**Comment by gokmen** [comment]
This is ready on it's own but I want to merge this in once the whole implementation ready.

**Review by sinan — APPROVED** [review]

File: workers/social/lib/social/models/oauth.coffee [hunk 1]
```diff
@@ -129,23 +131,83 @@ module.exports = class OAuth extends bongo.Base
     # -- GITHUB PROVIDER --------------------------------------------------8<--
 
     github    :
-      title   : 'Github OAuth Provider'
-      enabled : false
+      title   : 'GitHub OAuth Provider'
+      enabled : yes
+      scopes  : ['user', 'user:email', 'user:follow', 'public_repo', 'repo',
```
Comment (cihangir) [inline workers/social/lib/social/models/oauth.coffee]: some of the scopes here are redundant and i believe some other are not required, why did you add all the scopes?

Comment (cihangir) [inline workers/social/lib/social/models/oauth.coffee]: ok found the reason
